java/src/com/google/template/soy/xliffmsgplugin/XliffGenerator.java

```diff
@@ -90,63 +105,129 @@ static CharSequence generateXliff(
     ilb.increaseIndent();
 
     for (SoyMsg msg : msgBundle) {
+      generateSoyMsg(ilb, msg);
+    }
 
-      // Begin 'trans-unit'.
-      ilb.appendLineStart("<trans-unit id=\"", Long.toString(msg.getId()), "\"");
-      String contentType = msg.getContentType();
-      if (contentType != null && contentType.length() > 0) {
-        String xliffDatatype = CONTENT_TYPE_TO_XLIFF_DATATYPE_MAP.get(contentType);
-        if (xliffDatatype == null) {
-          xliffDatatype = contentType;  // just use the contentType string
-        }
-        ilb.appendParts(" datatype=\"", attributeEscaper.escape(xliffDatatype), "\"");
+    ilb.decreaseIndent();
+    ilb.appendLine("</body>");
+    ilb.decreaseIndent();
+    ilb.appendLine("</file>");
+    ilb.decreaseIndent();
+    ilb.appendLine("</xliff>");
+
+    return ilb;
+  }
+
+  private void generateSoyMsg(IndentedLinesBuilder ilb, SoyMsg msg) {
+    generateSoyMsgHelper(ilb, msg, msg.getParts(), "");
+  }
+
+  /**
+   * A message may be one or more nested select/plural branches.
+   *
+   * We need to flatten these into a list of transunits, one for each complete
+   * path through the branches.
+   */
+  private void generateSoyMsgHelper(IndentedLinesBuilder ilb, SoyMsg msg, List<SoyMsgPart> parts, String idSuffix) {
+    SoyMsgPart firstPart = Iterables.getFirst(parts, null);
+    if (firstPart instanceof SoyMsgSelectPart) {
+      if (parts.size() != 1) {
+        throw new IllegalStateException("Illegal SELECT AST");
       }
-      ilb.appendLineEnd(">");
+
+      ilb.appendLine("<group restype=\"" + XliffResType.SELECT + "\">");
```

Comment:
I'm not familiar with the xliff spec.  Why are selects and plurals done this way?